### ISSUE TITLE
UI not showing logs of ContainerSet containers

### ISSUE THREAD

**Issue by nice-pink** [argoproj/argo-workflows#10157]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

I'm running a workflow with a ContainerSet. When I want to see the logs of the containers I click on one container, then click on LOGS in sidebar menu. The logs window for all containers is empty but the main container.

I use `archiveLogs: true` in the workflow manifests, so I can also see the workflow in **Archived Workflows** tab. There, the logs are visible for each container.

### Version

v3.4.3

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: test
  namespace: argo
spec:
  archiveLogs: true
  activeDeadlineSeconds: 1200
  entrypoint: main
  serviceAccountName: workflow
  templates:
  - name: main
    metadata:
      labels:
        app: app
    containerSet:
      containers:
      - name: echo
        image: alpine:latest
        command: ["/bin/sh", "-c"]
        args:
        - |
          echo "echo"
      - name: main
        dependencies:
        - echo
        image: alpine:latest
        command: ["/bin/sh", "-c"]
        args:
        - |
          echo "main"
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

**Comment by terrytangyuan** [comment]
Any update on this?

**Comment by caelan-io** [comment]
@isubasinghe is working on new PRs for some containerSet improvements.

**Comment by nice-pink** [comment]
Is there an ETA?

**Comment by isubasinghe** [comment]
@nice-pink can you confirm for me that the logs are not appearing in the master branch or v3.5.4 ? 
This should be a solved issue I believe.

**Comment by agilgur5** [comment]
> Thanks for reporting this feature gap; it's a duplicate of #5316

Not quite, in the issue description, OP is saying for non-archived logs. As in, the logs of live Pods aren't showing

> I use `archiveLogs: true` in the workflow manifests, so I can also see the workflow in **Archived Workflows** tab. There, the logs are visible for each container.

From OP, the archived logs are showing up.

**Comment by caelan-io** [comment]
Got it, thanks for clarifying, Anton. 

@isubasinghe - can you please check this out for "live pods with containersets"? I guess that's the scenario where logs aren't showing. If that's still the case on v3.5, I'm in support of shipping it. 

Curious if this is a small enough feature/ fix to fit into a 3.5 patch, or if ya'll think we should save for the 3.6 release.

**Comment by agilgur5** [comment]
> Curious if this is a small enough feature/ fix to fit into a 3.5 patch, or if ya'll think we should save for the 3.6 release.

it's in between the two and only impacts containerSets, so I think it'd be fine for a patch IMO

**Comment by isubasinghe** [comment]
> Got it, thanks for clarifying, Anton.
> 
> @isubasinghe - can you please check this out for "live pods with containersets"? I guess that's the scenario where logs aren't showing. If that's still the case on v3.5, I'm in support of shipping it.
> 
> Curious if this is a small enough feature/ fix to fit into a 3.5 patch, or if ya'll think we should save for the 3.6 release.

Can confirm that I can still see logs in running container sets.

**Comment by agilgur5** [comment]
Superseded by #13038 which found the specific issue
